src/query/api/v1/handler/namespace/add_test.go

```diff
@@ -203,40 +203,56 @@ func TestNamespaceAddHandler_Conflict(t *testing.T) {
 	assert.Equal(t, http.StatusConflict, resp.StatusCode)
 }
 
-func TestValidateNewMetadata(t *testing.T) {
-	// Valid.
-	addReq := new(admin.NamespaceAddRequest)
-	require.NoError(t, jsonpb.Unmarshal(strings.NewReader(testAddJSON), addReq))
-	md, err := namespace.ToMetadata(addReq.Name, addReq.Options)
-	require.NoError(t, err)
-	require.NoError(t, validateNewMetadata(md))
-
-	// Valid without index options.
-	addReq = new(admin.NamespaceAddRequest)
-	require.NoError(t, jsonpb.Unmarshal(strings.NewReader(testAddJSON), addReq))
-	addReq.Options.RetentionOptions.BlockSizeNanos = 7200000000000 / 2
-	addReq.Options.IndexOptions = nil
-	md, err = namespace.ToMetadata(addReq.Name, addReq.Options)
-	require.NoError(t, err)
-	require.NoError(t, validateNewMetadata(md))
-
-	// Invalid without retention options.
-	addReq = new(admin.NamespaceAddRequest)
-	require.NoError(t, jsonpb.Unmarshal(strings.NewReader(testAddJSON), addReq))
-	addReq.Options.RetentionOptions = nil
-	addReq.Options.IndexOptions.BlockSizeNanos = 7200000000000 / 2
-	md, err = namespace.ToMetadata(addReq.Name, addReq.Options)
-	require.Error(t, err)
-	require.Equal(t, "retention options must be set", err.Error())
-
-	// Prevent mismatching block sizes.
-	addReq = new(admin.NamespaceAddRequest)
-	require.NoError(t, jsonpb.Unmarshal(strings.NewReader(testAddJSON), addReq))
-	addReq.Options.RetentionOptions.BlockSizeNanos = 7200000000000
-	addReq.Options.IndexOptions.BlockSizeNanos = 7200000000000 * 2
-	md, err = namespace.ToMetadata(addReq.Name, addReq.Options)
-	require.NoError(t, err)
-	err = validateNewMetadata(md)
-	require.Error(t, err)
-	require.Equal(t, "index and retention block size must match (2h0m0s, 4h0m0s)", err.Error())
+func TestApplyNewNamespaceValidator(t *testing.T) {
```

Comment:
Not sure about the name of the test. Kind of expects success but in the test itself, we do expect a bad request? In anyway maybe it's worth to name it in the way why we should expect it to fail.